experimental/ir/fdp.go

```diff
@@ -24,92 +25,84 @@ import (
+// The resulting FileDescriptorSet is always fully linked: it contains all dependencies except
 // the WKTs, and all names are fully-qualified.
-func FDS(files []File, options ...CodegenOption) ([]byte, error) {
-	var c codegen
+func DescriptorSet(files []File, options ...DescriptorOption) ([]byte, error) {
+	var dg descGenerator
 	for _, opt := range options {
 		if opt != nil {
-			opt(&c)
+			opt(&dg)
 		}
 	}
 
 	fds := new(descriptorpb.FileDescriptorSet)
-	c.files(files, fds)
+	dg.files(files, fds)
 	return proto.Marshal(fds)
 }
 
-// FDP code-generates a single FileDescriptorProto for file, and returns the
+// DescriptorProto generates a single FileDescriptorProto for file, and returns the
 // result as an encoded byte slice.
 //
-// The resulting FDP is fully linked: all names are fully-qualified.
-func FDP(file File, options ...CodegenOption) ([]byte, error) {
-	var c codegen
+// The resulting FileDescriptorProto is fully linked: all names are fully-qualified.
+func DescriptorProto(file File, options ...DescriptorOption) ([]byte, error) {
+	var dg descGenerator
 	for _, opt := range options {
 		if opt != nil {
-			opt(&c)
+			opt(&dg)
 		}
 	}
 
 	fdp := new(descriptorpb.FileDescriptorProto)
-	c.file(file, fdp)
+	dg.file(file, fdp)
 	return proto.Marshal(fdp)
 }
 
-// CodegenOption is an option to pass to [DescriptorSet].
-type CodegenOption func(*codegen)
+// DescriptorOption is an option to pass to [DescriptorSet] or [DescriptorProto].
+type DescriptorOption func(*descGenerator)
 
-// IncludeWKTs is an option for including FDPs for the Well-Known Types in
-// codegen output.
-func IncludeWKTs() CodegenOption {
-	return func(c *codegen) { c.includeWKTs = true }
+// ExcludeFiles is an option that causes all files that match the given
+// predicate to be excluded from the output of [DescriptorSet].
+func ExcludeFiles(p func(File) bool) DescriptorOption {
+	return func(dg *descGenerator) { dg.exclude = append(dg.exclude, p) }
 }
 
-// IncludeSourceCodeInfo is an option that requires generating source code info
-// for the descriptors.
-func IncludeSourceCodeInfo() CodegenOption {
-	return func(c *codegen) { c.includeSourceCodeInfo = true }
-}
-
-type codegen struct {
-	includeWKTs           bool
-	includeSourceCodeInfo bool
+type descGenerator struct {
+	exclude []func(File) bool
 
 	currentFile File
 }
 
-func (c *codegen) files(files []File, fds *descriptorpb.FileDescriptorSet) {
-	if c.includeSourceCodeInfo {
-		panic("protocompile/ir: NYI: IncludeSourceCodeInfo()")
-	}
-
+func (dg *descGenerator) files(files []File, fds *descriptorpb.FileDescriptorSet) {
 	// Build up all of the imported files. We can't just pull out the transitive
 	// imports for each file because we want the result to be sorted
 	// topologically.
-	for file := range TopoSort(files...) {
-		if !c.includeWKTs && wellknownimports.StandardImport(file.Path()) != "" {
-			continue
+	for file := range topoSort(files...) {
+		for _, p := range dg.exclude {
```

Comment:
This would be more readable (particularly the next line) to name it `exclude` instead of `p`.